### ISSUE TITLE
New "Feature": mounting a fake home

### ISSUE THREAD

**Issue by marcodelapierre** [singularityhub/singularity-hpc#434]
There are some quite significant user cases, where an application needs to write some small files in order to run (otherwise it would crash), typically in the user home directory.

Amongst the examples: some Python applications, some Conda environments, some Java applications (most recent example: the Beast2 package from quay.io/biocontainers, that is also available in SHPC through Autamus).

This is mostly a problem with Singularity and its read-only filesystem; I know of some Singularity deployments (eg ours at Pawsey) where mounting home is disabled by default as a best practices.

Proposed solution: allow the registry to have a feature to mount a fake home, provided as customary path: `-B <host-fake-home-path>:$HOME` (alternatively, it could just be using `--writable-tmpfs`).

- Config: `home: <path-to-host-fake-home>` (might suggest a reasonable choice in the comment such as `~/.shpc_home`)
- Registry: `home: true` (default false)

What do you think?

**Comment by marcodelapierre** [comment]
Similar to #433, suggested idea for (fake) home feature:

SHPC settings:

```
container_features:
  fake_home: "<path-to-fake-home>" # a good default here might be simply "$HOME", but some sites might use a truly fake home such as "$HOME/.shpc_home"
```

Package recipe:
```
features:
  fake_home: true  # or false (default)
```

Implementation: when feature is "true" in the package recipe, and container_feature is different from null: add a exec/run/shell flag:
- Singularity: -B <value of fake_home in settings>:$HOME
- Docker/Podman: -v <value of fake_home in settings>:$HOME

Note it will be up to sys admin or users to create the fake_home path in the host, in case it's different from $HOME

**Comment by vsoch** [comment]
Wouldn't we want to use `--home` for Singularity too?

**Comment by vsoch** [comment]
Closed with #444
